### ISSUE TITLE
What is the goal?

### ISSUE THREAD

**Issue by biggzlar** [isl-org/DirectFuturePrediction#8]
https://github.com/IntelVCL/DirectFuturePrediction/blob/b4757769f167f1bd7fb1ece5fdc6d874409c68a9/DFP/future_predictor_agent_advantage.py#L86

Maybe this is just me, but I find this section as described in the paper highly confusing. What exactly is **g**? Is it the objective coefficients, the temporal coefficients (since it is supposed to have the same dimensionality as **f**, not as **f_i**), a combination of the two as this implementation assumes or the actual **g * f**?

**Comment by avijit9** [comment]
@biggzlar  did you figure it out?

**Comment by biggzlar** [comment]
@avijit9 Hey there, I'm afraid not. Sorry. :/

**Comment by avijit9** [comment]
I think I have figured it out. Check the `make_loss` function
`per_target_loss = my_ops.mse_ignore_nans(pred_relevant, targets_preprocessed, reduction_indices=0)
loss = tf.reduce_sum(per_target_loss)`

That means you take mean across samples (as given in equation (4) and then just take sum across the time-steps. You have temporal coefficients `(0,0,0, 0.5, 0.5, 1)` and goal defines your objective.
